File: Laboratorio/oct1/part4/Led.py
from machine import Pin
import time

pin_led = None

def iniciar_led(numero_pin):
    global pin_led
    pin_led = Pin(numero_pin, Pin.OUT)
    apagar_led()

def encender_led():
    pin_led.value(1)

def apagar_led():
    pin_led.value(0)

def parpadear_led(frecuencia_hz):
    if frecuencia_hz <= 0:
        apagar_led()
        return
    medio_periodo_ms = int(500 / frecuencia_hz)
    encender_led()
    time.sleep_ms(medio_periodo_ms)
    apagar_led()
    time.sleep_ms(medio_periodo_ms)

File: Laboratorio/oct1/part2/src/Led.cpp
#include "Led.h"
uint8_t pinLed;

void iniciarLed(uint8_t pin)
{
    pinLed = pin;
    pinMode(pinLed, OUTPUT);
    apagarLed();
}

void encenderLed()
{
    digitalWrite(pinLed, HIGH);
}

void apagarLed()
{
    digitalWrite(pinLed, LOW);
}

void parpadearLed(float frecuenciaHz)
{
    if (frecuenciaHz <= 0.0)
    {
        apagarLed();
        return;
    }
    unsigned long medioPeriodoMs = 500.0 / frecuenciaHz;
    encenderLed();
    delay(medioPeriodoMs);
    apagarLed();
    delay(medioPeriodoMs);
}